topic/fix-course-attendanceをdevelopに対してプルリク作成 講座の受講期限関連のブランチをマージ

yossy691 opened the pull request.

satoko567 committed:
講座の受講期限を「受講開始から⚪︎ヶ月有効」なども対応できるようにテーブル変更・作成

satoko567 committed:
テーブル設計の修正

satoko567 committed:
fix: course_deadlinesとcourses テーブル・シーダーの修正

Commit:
feat: マイグレーションファイルを修正。

kou-tech committed:
Merge pull request #1062 from yukihiroLaravel/feature/chaco/jka-1547/course-expiration-schema

講座の受講期限を「受講開始から⚪︎ヶ月有効」なども対応できるようにテーブル変更・作成(ちゃこ)

satoko567 committed:
講師側 講座登録APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応

satoko567 committed:
講師側 講座更新APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応

satoko567 committed:
JKA-1555: 講座登録APIのコントローラ／モデル微修正

satoko567 committed:
Courseモデルの修正

satoko567 committed:
Manager/NotificationController.phpの差分戻し

miiwa-5030 committed:
JKA-1550 講師側 講座編集(詳細)APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応

tatsunori-1550 committed:
WIP作業中断

satoko567 committed:
マネージャー側 講座登録APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応

Commit:
Refactor: 不要なuse文を削除し、StoreCourseServiceの引数を整形

Commit:
Merge branch 'topic/fix-course-attendance' into feature/tomomi/jka-1550

Commit:
Refactor: 不要なインポートと空白を削除

Commit:
Refactor: 不要な空白を削除

Commit:
Refactor: コードの可読性向上のために空白を追加

Morio151515 committed:
JKA-1552-1

Morio151515 committed:
Merge branch 'develop' into feature/mitsunobu/JKA-1552/course-edit-show

Morio151515 committed:
JKA-1552-2

Morio151515 committed:
JKA-1552-3

Morio151515 committed:
JKA-1552-3

Morio151515 committed:
JKA-1552-3

Morio151515 committed:
JKA-1552-3

Morio151515 committed:
JKA-1552-3

Morio151515 committed:
JKA-1552-3

Morio151515 committed:
JKA-1552-3

Morio151515 committed:
JKA-1552-3

Morio151515 committed:
JKA-1552-5

satoko567 committed:
JKA-1553: 講座更新APIのEnumバリデーション対応

satoko567 committed:
JKA-1555: DeadlineTypeEnum の導入に伴うバリデーション・サービス・コントローラの対応

Commit:
Refactor: StoreCourseServiceをStoreServiceに変更し、講座登録APIの処理を整理

Commit:
Add: 講座登録APIに受講期限の設定を追加し、関連するサービスとテストを実装

kou-tech committed:
Merge pull request #1065 from yukihiroLaravel/feature/chaco/jka-1555/course-deadline-type-option

講師側 講座登録APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応(ちゃこ)

satoko567 committed:
JKA-1553: hasDeadline処理をプライベート関数として切り出し

satoko567 committed:
Course/StoreService.phpのコンフリクト解消

satoko567 committed:
JKA-1556: DeadlineTypeEnum 対応（Controller/Request/Service）

satoko567 committed:
[jka-1560] お知らせ一覧：受講期限切れの講座を除外する対応

miiwa-5030 committed:
JKA-1550 講師側 講座編集(詳細)APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応 修正

satoko567 committed:
マイグレーションファイルのファイル名変更

satoko567 committed:
マイグレーションファイルのファイル名変更

satoko567 committed:
マイグレーションファイルのファイル名変更

tatsunori-1550 committed:
受講生側 講座一覧 講座の受講期限の新設定「受講開始からヶ月有効」などを考慮して対応

Commit:
Merge remote-tracking branch 'origin/topic/fix-course-attendance' into feature/chaco/jka-1553/course-deadline-update-api

Commit:
Remove: 不要なモデルとマイグレーションファイルを削除し、コードのクリーンアップを実施

Commit:
Refactor: UpdateServiceとCourseControllerのサービス名を変更し、関連メソッド名を修正

kou-tech committed:
Merge pull request #1067 from yukihiroLaravel/feature/chaco/jka-1553/course-deadline-update-api

講師側 講座更新APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応(ちゃこ)

Commit:
Merge remote-tracking branch 'origin/topic/fix-course-attendance' into feature/chaco/jka-1556/manager-course-deadline-type-option

Morio151515 committed:
WIP: local changes before merging base branch

Commit:
Add: 講座登録機能に受講期限の設定を追加し、関連するバリデーションとマイグレーションを実装

kou-tech committed:
Merge pull request #1070 from yukihiroLaravel/feature/chaco/jka-1556/manager-course-deadline-type-option

マネージャー側 講座登録APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応(ちゃこ)

Commit:
Merge remote-tracking branch 'origin/topic/fix-course-attendance' into feature/tatsunori/jka-1548/student-course-list-deadline-display

Commit:
Refactor: コードの整形と不要なインポートの削除

Morio151515 committed:
WIP: include CourseDeadline changes

Morio151515 committed:
Resolve merge: use theirs from topic/fix-course-attendance

Morio151515 committed:
fix: add Exception import in NotificationController

Commit:
Merge branch 'topic/fix-course-attendance' into feature/tomomi/jka-1550

Commit:
Add: 講座の締切情報を追加し、関連するリソースとシーダーを実装

Commit:
Remove: 不要なレスポンスダンプを削除

kou-tech committed:
Merge pull request #1068 from yukihiroLaravel/feature/tomomi/jka-1550

講師側 講座編集(詳細)APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応(智美)

Commit:
Merge remote-tracking branch 'origin/topic/fix-course-attendance' into feature/chaco/jka-1560/exclude-expired-course-notifications

miiwa-5030 committed:
JKA-1554 マネージャー側 講座更新APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応

Morio151515 committed:
refactor(resource): move deadline to CourseDeadlineResource & include conditionally

Morio151515 committed:
JKA-1552-6

Morio151515 committed:
JKA-1552-6

Morio151515 committed:
JKA-1552-6

Commit:
Add: 講座の締切情報をAttendanceモデルに追加し、関連するマイグレーションとシーダーを実装

kou-tech committed:
Merge pull request #1072 from yukihiroLaravel/feature/chaco/jka-1560/exclude-expired-course-notifications

お知らせ一覧：受講期限切れの講座を除外する対応(ちゃこ)

Morio151515 committed:
JKA-1552-7

Morio151515 committed:
Resolve conflicts: adopt topic/fix-course-attendance for Course* resources

satoko567 committed:
JKA-1559 マネージャー側 講座一覧APIに受講期限表示を追加

Commit:
Add: 講座更新テストに受講期限の設定を追加し、関連するアサーションを実装

Morio151515 committed:
JKA-1552-7

tatsunori-1550 committed:
レビュー修正

tatsunori-1550 committed:
コンフリクト修正

satoko567 committed:
CourseControllerとCourseIndexResourceの修正（CourseDeadlineResource利用）

Morio151515 committed:
JKA-1552-8

Commit:
Change: 更新日時のプロパティから出席締切を削除

kou-tech committed:
Merge pull request #1071 from yukihiroLaravel/feature/mitsunobu/JKA-1552/course-edit-show

マネージャー側 講座編集(詳細)APIの修正 講座の受講期限の新設定「受講開始から◯ヶ月有効」などを考慮して対応（もりお）

Commit:
Merge remote-tracking branch 'origin/topic/fix-course-attendance' into feature/chaco/jka-1559/manager-course-list-deadline-display

satoko567 committed:
受講生側 お知らせ既読登録APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応

miiwa-5030 committed:
JKA-1562 マネージャー側 受講状況取得APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応

miiwa-5030 committed:
JKA-1554 マネージャー側 講座更新APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応 修正

miiwa-5030 committed:
Merge branch 'feature/tomomi/jka-1554' of https://github.com/yukihiroLaravel/juko_laravel into feature/tomomi/jka-1554

miiwa-5030 committed:
Merge branch 'topic/fix-course-attendance' into feature/tomomi/jka-1562

Commit:
Remove: CourseIndexResourceから締切情報を削除

kou-tech committed:
Merge pull request #1075 from yukihiroLaravel/feature/chaco/jka-1559/manager-course-list-deadline-display

マネージャー側 講座一覧APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応(ちゃこ)

Commit:
Refactor: コードの順序を整理し、CourseStatusRuleのインポートを修正

tatsunori-1550 committed:
各レビュー修正

satoko567 committed:
fix: attendance_deadline を利用した受講期限チェックに修正（レビュー対応）

satoko567 committed:
講師・マネージャー側 受講状況APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応

kou-tech committed:
Merge pull request #1074 from yukihiroLaravel/feature/tomomi/jka-1554

マネージャー側 講座更新APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応(智美)

Commit:
Merge branch 'topic/fix-course-attendance' into feature/tatsunori/jka-1548/student-course-list-deadline-display

Commit:
Remove: 不要なマイグレーションファイルを削除

Commit:
Refactor: 不要な改行を削除

tatsunori-1550 committed:
不要箇所修正

tatsunori-1550 committed:
Merge branch 'feature/tatsunori/jka-1548/student-course-list-deadline-display' of https://github.com/yukihiroLaravel/juko_laravel into feature/tatsunori/jka-1548/student-course-list-deadline-display

miiwa-5030 committed:
JKA-1562 マネージャー側 受講状況取得APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応

Morio151515 committed:
JKA-1552-1

Morio151515 committed:
Merge branch 'topic/fix-course-attendance' into feature/mitsunobu/JKA-1566/fix-student-notification-detail

Commit:
Refactor: 不要なコードを削除し、可読性を向上

Commit:
Refactor: isExpiredメソッドの条件式を簡略化

miiwa-5030 committed:
JKA-1562 マネージャー側 受講状況取得APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応 修正

tatsunori-1550 committed:
Attendance期限処理をアクセサに統一

satoko567 committed:
fix: 明示的に null チェックを追加（レビュー対応）

satoko567 committed:
fix: attendance_deadline を immutable_datetime でキャスト & 期限切れチェックを修正

satoko567 committed:
feat(manager): attendance_deadline算出をサービスへ切り出し＆storeで適用（showは変更なし）

Commit:
Refactor: 不要なコードを削除し、リソースの構造を整理

Commit:
Merge branch 'topic/fix-course-attendance' into feature/tomomi/jka-1562

Morio151515 committed:
Student Notification detail: add ShowService and expiry check (individual > fixed > relative > none)

Commit:
Fix: 'courseDeadline'を'course_deadline'に修正

Commit:
Refactor: 受講状況取得APIのリファクタリングとバリデーションの修正

Commit:
Change: 'attendance_type'を'deadline_type'に修正

Commit:
Remove: 不要なCarbonImmutableのインポートとgetFixedDeadlineEndAttributeメソッドを削除

kou-tech committed:
Merge pull request #1078 from yukihiroLaravel/feature/tomomi/jka-1562

マネージャー側 受講状況取得APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応(智美)

Commit:
Merge remote-tracking branch 'origin/topic/fix-course-attendance' into feature/tatsunori/jka-1548/student-course-list-deadline-display

Commit:
Change: Attendanceポリシーの閲覧権限を修正し、バリデーションを簡素化

kou-tech committed:
Merge pull request #1073 from yukihiroLaravel/feature/tatsunori/jka-1548/student-course-list-deadline-display

受講生側 講座一覧 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応 (たつのり)

Commit:
Merge remote-tracking branch 'origin/topic/fix-course-attendance' into feature/chaco/jka-1567/student-notification-markread-error-expired

Commit:
Change: Attendanceモデルのインポート位置を修正し、不要なインポートを削除

Commit:
Change: CourseモデルをAttendanceモデルに置き換え、期限切れのテストを修正

Commit:
Merge remote-tracking branch 'origin/topic/fix-course-attendance' into feature/chaco/jka-1563/manager-attendance-deadline-response

satoko567 committed:
refactor: Attendance::isExpired() を利用して期限切れ判定を共通化（レビュー対応）

satoko567 committed:
refactor(manager): 受講期限計算を__invokeサービス化＆メソッドDI化。now基準で算出しcreate時に同時保存

Morio151515 committed:
JKA-1566-1

Morio151515 committed:
Merge latest topic/fix-course-attendance

Morio151515 committed:
WIP: sync before merging fix-course-attendance

Morio151515 committed:
WIP: sync before merging fix-course-attendance

Commit:
Add: 受講期限を算出するCalculateDeadlineServiceを追加し、AttendanceControllerで使用

Commit:
Change: Attendanceモデルのcreated_atとupdated_atの型をimmutable_datetimeに変更

Commit:
Refactor: AttendanceControllerのstoreメソッドでCalculateDeadlineServiceの引数名を変更

Morio151515 committed:
JKA-1566-2

miiwa-5030 committed:
JKA-1572 マネージャー側 講座削除APIを修正して、講座が削除されたらその講座に紐付くCourseDeadlineも削除されるように対応

satoko567 committed:
fix: Attendance レコード未存在時は例外を投げるよう修正（レビュー対応）

Morio151515 committed:
JKA-1566-3

Commit:
Remove: 講座削除APIと関連するリクエスト、サービスを削除し、講座締切データの削除処理を修正

kou-tech committed:
Merge pull request #1081 from yukihiroLaravel/feature/tomomi/jka-1572

マネージャー側 講座削除APIを修正して、講座が削除されたらその講座に紐付くCourseDeadlineも削除されるように対応(智美)

Commit:
Change: お知らせ既読登録成功のテストメソッド名を修正

kou-tech committed:
Merge pull request #1076 from yukihiroLaravel/feature/chaco/jka-1567/student-notification-markread-error-expired

受講生側 お知らせ既読登録APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応(ちゃこ)

Commit:
Merge remote-tracking branch 'origin/topic/fix-course-attendance' into feature/mitsunobu/JKA-1566/fix-student-notification-detail

Commit:
Change: NotificationControllerにShowServiceのインポートを追加し、ShowTestのテストメソッド名を修正

Commit:
Change: ShowTestのテストメソッド名を修正

kou-tech committed:
Merge pull request #1079 from yukihiroLaravel/feature/mitsunobu/JKA-1566/fix-student-notification-detail

受講生側 お知らせ詳細APIの修正 講座の受講期限の新設定「受講開始から◯ヶ月有効」などを考慮して対応（もりお）

satoko567 committed:
feat(manager): CalculateDeadlineService追加 / CalculateDeadlineServiceTest正常系テスト追加

miiwa-5030 committed:
JKA-1574 マネージャー側 「全て受講期限をなくす」APIの新規作成

miiwa-5030 committed:
JKA-1574 マネージャー側 「全て受講期限をなくす」APIの新規作成 コメントアウトの修正

miiwa-5030 committed:
JKA-1574 マネージャー側 「全て受講期限をなくす」APIの新規作成 コメントアウトの修正2

Commit:
Refactor: CalculateDeadlineServiceの例外メッセージを改善し、テストケースを追加

Commit:
Merge branch 'topic/fix-course-attendance' into feature/chaco/jka-1563/manager-attendance-deadline-response

kou-tech committed:
Merge pull request #1077 from yukihiroLaravel/feature/chaco/jka-1563/manager-attendance-deadline-response

講師・マネージャー側 受講状況APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応(ちゃこ)

miiwa-5030 committed:
JKA-1574 マネージャー側 「全て受講期限をなくす」APIの新規作成 コメントアウトの修正

miiwa-5030 committed:
JKA-1574 マネージャー側 「全て受講期限をなくす」APIの新規作成 修正

Commit:
Add: 「全て受講期限をなくす」APIの実装とテストを追加

kou-tech committed:
Merge pull request #1083 from yukihiroLaravel/feature/tomomi/jka-1574

マネージャー側 「全て受講期限をなくす」APIの新規作成(智美)

YoshikatsuInoue committed:
JKA-1558 講師側 講座一覧APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応

Commit:
Merge branch 'topic/fix-course-attendance' into feature/inocchi/jka-1558/course-expiration

Commit:
Refactor: TagResourceのインポート順を修正

kou-tech committed:
Merge pull request #1087 from yukihiroLaravel/feature/inocchi/jka-1558/course-expiration

JKA-1558 講師側 講座一覧APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応(いのっち)

yossy691 committed:
講師側 お知らせ一覧取得APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応(なかよし)

yossy691 committed:
講師側 お知らせ一覧取得APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応(なかよし)

yossy691 committed:
講師側 お知らせ一覧取得APIの修正 講座の受講期限の新設定「受講開始から:白い丸:︎ヶ月有効」などを考慮して対応(なかよし)

yossy691 committed:
講師側 お知らせ一覧取得APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応

yossy691 committed:
講師側 お知らせ一覧取得APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応

yossy691 committed:
講師側 お知らせ一覧取得APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応(なかよし)

kou-tech committed:
Merge pull request #1092 from yukihiroLaravel/feature/fix-instructor-notifications-expiration

講師側 お知らせ一覧取得APIの修正 講座の受講期限の新設定「受講開始から⚪︎ヶ月有効」などを考慮して対応

Commit:
Merge branch 'develop' into topic/fix-course-attendance

Commit:
Fix: AttendancePolicyの期限判定の論理反転を修正（viewStudent, view）およびStoreTestのfixed_date期待値を日時フォーマットに更新

kou-tech commented:
@yossy691 
こちらのエラーメッセージの対応をお願いします！
講師のタグ一覧取得APIの改修
対象ファイル：app/Http/Controllers/Api/Instructor/Course/TagController.phpのindexメソッド

```txt
Attempted to lazy load [courseDeadline] on model [App\Model\Course] but lazy loading is disabled.
```

Commit:
Test: UpdateTestのfixed_date期待値を日時フォーマット('Y-m-d 00:00:00')に修正、StoreTestから期限切れ講座テストと未使用のCourse importを削除

kou-tech commented:
@yossy691 
講師-講座情報一覧取得APIの改修
対象ファイル：app/Http/Controllers/Api/Manager/Instructor/CourseController.phpのindexメソッド

```txt
Attempted to lazy load [courseDeadline] on model [App\Model\Course] but lazy loading is disabled.
```

Commit:
wip

kou-tech commented:
`app/Http/Controllers/Api/Manager/Instructor/InstructorController.php`のindexメソッド

```txt
----------------------------------------------------------------------------------

SQLSTATE[HY000]: General error: 1 no such column: courses.attendance_deadline (Connection: sqlite, SQL: select "instructors".*, (select COUNT(DISTINCT attendances.student_id) from "courses" inner join "attendances" on "courses"."id" = "attendances"."course_id" where "instructors"."id" = "courses"."instructor_id" and ("courses"."attendance_deadline" is null or "courses"."attendance_deadline" > 2025-10-22 19:20:06) and "courses"."deleted_at" is null) as "student_count" from "instructors" where "id" in (2, 3) order by "email" desc limit 20 offset 0)

  at tests/Feature/Api/Manager/Instructor/IndexTest.php:32
     28▕         // act
     29▕         $response = $this->getJson('/api/v1/manager/instructor/index');
     30▕
     31▕         // assert
  ➜  32▕         $response->assertStatus(200);
     33▕         $response->assertJsonCount(2, 'data.instructors');
     34▕         $response->assertJson([
     35▕             'data' => [
     36▕                 'instructors' => [

  ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   FAILED  Tests\Feature\Api\Manager\Instructor\IndexTest > 期限切れの講座が存在 講師講座一覧取得 成功                                                                                                    MassAssignmentException
  Add fillable property [attendance_deadline] to allow mass assignment on [App\Model\Course].

  at vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php:559
    555▕
    556▕             if (isset(static::$discardedAttributeViolationCallback)) {
    557▕                 call_user_func(static::$discardedAttributeViolationCallback, $this, $keys);
    558▕             } else {
  ➜ 559▕                 throw new MassAssignmentException(sprintf(
    560▕                     'Add fillable property [%s] to allow mass assignment on [%s].',
    561▕                     implode(', ', $keys),
    562▕                     get_class($this)
    563▕                 ));

      +2 vendor frames
  3   tests/Feature/Api/Manager/Instructor/IndexTest.php:66
```

yossy691 committed:
topic/fix-course-attendanceをdevelopに対してプルリク作成 講座の受講期限関連のブランチをマージ

kou-tech commented:
@yossy691 
- `app/Http/Controllers/Api/Instructor/Course/TagController.php`のindexメソッドの改修
- `app/Http/Controllers/Api/Manager/TagController.php`のindexメソッドの改修


```txt
Attempted to lazy load [courseDeadline] on model [App\Model\Course] but lazy loading is disabled.
```

kou-tech commented:
@yossy691 

`app/Policies/NotificationPolicy.php`のstoreメソッド(お知らせの作成に関する認可処理)受講期限チェックの削除。

```txt
The attribute [attendance_deadline_end] either does not exist or was not retrieved for model [App\Model\Course].
```

yossy691 committed:
topic/fix-course-attendanceをdevelopに対してプルリク作成 講座の受講期限関連のブランチをマージ

Commit:
Change: 受講期限周りの修正とテスト更新、不要 import と依存更新

- UpdateTest の fixed_date 期待値を日時フォーマット('Y-m-d 00:00:00')に修正
- IndexTest/StoreTest/ProgressTest を受講期限モデルに合わせて修正（Course→Attendance参照、一部期限切れテストの削除/調整）
- NotificationPolicy の未使用 import を削除
- InstructorController の不要な空白を削除
- composer.lock の依存バージョンを更新
